### PR TITLE
python3Packages.flitBuildHook: remove

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

This change removes all instances of `format = "flit";`, and removes `flitBuildHook`.
This backward-seeming order is to be git-bisect friendly. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
This will likely break many of the overrides defined in poetry2nix

**Review by dotlambda — DISMISSED** [review]
Note the typo in the commit message.

**Comment by pbsds** [comment]
I think I need some pointers on the eval error. I'm not even sure how to reproduce it locally.

**Comment by figsoda** [comment]
The eval error looks like nixops_unstable, which uses poetry2nix

File: pkgs/development/python-modules/zeversolarlocal/default.nix [hunk 1]
```diff
@@ -21,6 +22,19 @@ buildPythonPackage rec {
     sha256 = "ExZy5k5RE7k+D0lGmuIkGWrWQ+m24K2oqbUEg4BAVuY=";
   };
 
+  # patch below won't apply due to different line endings
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/zeversolarlocal/default.nix:26]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/zeversolarlocal/default.nix [hunk 2]
```diff
@@ -21,6 +22,19 @@ buildPythonPackage rec {
...
+    sed -e "s/\r//g" -i pyproject.toml
+  '';
+
+  patches = [
+    # Raise the flit-core limit
+    (fetchpatch { # https://github.com/sander76/zeversolarlocal/pull/4
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/zeversolarlocal/default.nix:32]:
```suggestion
    # https://github.com/sander76/zeversolarlocal/pull/4
    (fetchpatch {
```

File: pkgs/development/python-modules/pytest-cid/default.nix [hunk 3]
```diff
@@ -25,6 +26,10 @@ buildPythonPackage rec {
       --replace "pytest >= 5.0, < 7.0" "pytest >= 5.0"
   '';
 
+  nativeBuildInputs = [
+    flit
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-cid/default.nix:30]: same here, why not use flit-core?

File: pkgs/development/python-modules/solo-python/default.nix [hunk 4]
```diff
@@ -26,6 +27,10 @@ buildPythonPackage rec {
     hash = "sha256-XVPYr7JwxeZfZ68+vQ7a7MNiAfJ2bvMbM3R1ryVJ+OU=";
   };
 
+  nativeBuildInputs = [
+    flit
```
[the remark this answers is not in the document]
Comment (pbsds) [reply pkgs/development/python-modules/solo-python/default.nix:31]: https://github.com/solokeys/solo1-cli/blob/5b0fde3ee3d48e48a3acb83c6f21ae7f0abcffc6/pyproject.toml#L2C1-L2C1

File: pkgs/development/python-modules/zeversolarlocal/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/python-modules/zeversolarlocal/default.nix:26]: This needs to run before `patchPhase` where `patches` are applied

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 5]
```diff
@@ -290,3 +290,5 @@ The module update takes care of the new config syntax and the data itself (user
   ./common/auto-format-root-device.nix ];` When you use the systemd initrd, you
   can automatically format the root device by setting
   `virtualisation.fileSystems."/".autoFormat = true;`.
+
+- `python3.pkgs.flitBuildHook` has been removed. Use `flit-core` and `format = "pyproject"` instead.
```
[the remark this answers is not in the document]
Comment (pbsds) [reply nixos/doc/manual/release-notes/rl-2311.section.md:294]: Please elaborate

File: pkgs/development/python-modules/pytest-cid/default.nix (hunk 3 above)
Comment (pbsds) [reply pkgs/development/python-modules/pytest-cid/default.nix]: Must've crossed my debugging

**Comment by pbsds** [comment]
This force-push does a major find-replace in `poetry2nix` in `poetry2nix: remove flitBuildHook`, replacing `flitBuildHook` with `flit-core`. Who can i @ to have a look at this?

**Comment by pbsds** [comment]
Full disclosure, I just discovered #251880, a duplicate effort. I thought it was odd how every time i needed a patch, @tjni seemed to have just submitted one to the upstream. I tried searching for whether he had a nixpkgs pull, but for some reason it did not show up with my search terms. His work also targets additional issues such as darwin compat, while this PR stricly removes flitBuildHook only.

File: pkgs/development/python-modules/solo-python/default.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (FRidh) [reply pkgs/development/python-modules/solo-python/default.nix:31]: Preferred solution here is to patch the pyproject.toml.

**Comment by FRidh** [comment]
Unmerged changes in https://github.com/NixOS/nixpkgs/pull/251880 can be rebased later.

**Comment by pbsds** [comment]
This pull may have triggered a build failiure in tifffile, addressed in #257267

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2311.section.md:339]: changelog sections should be randomly sorted and not appeneded

**Pull request merged**